hadoop-hdds/server-scm/src/main/java/org/apache/hadoop/hdds/scm/metadata/SCMMetadataStoreMetrics.java

```diff
@@ -18,46 +18,39 @@
 package org.apache.hadoop.hdds.scm.metadata;
 
 import org.apache.commons.text.WordUtils;
-import org.apache.hadoop.hdds.utils.db.DBColumnFamilyDefinition;
+import org.apache.hadoop.hdds.utils.db.Table;
 import org.apache.hadoop.metrics2.MetricsCollector;
 import org.apache.hadoop.metrics2.MetricsInfo;
 import org.apache.hadoop.metrics2.MetricsRecordBuilder;
 import org.apache.hadoop.metrics2.MetricsSource;
 import org.apache.hadoop.metrics2.annotation.Metrics;
 import org.apache.hadoop.metrics2.lib.DefaultMetricsSystem;
-import org.apache.hadoop.metrics2.lib.Interns;
 import org.apache.hadoop.metrics2.lib.MetricsRegistry;
 import org.apache.hadoop.ozone.OzoneConsts;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
 
 import java.io.IOException;
-import java.util.Collections;
-import java.util.LinkedHashMap;
 import java.util.Map;
 
+import static org.apache.hadoop.metrics2.lib.Interns.info;
+
 /**
  * Class contains metrics related to SCM Metadata Store.
  */
 @Metrics(about = "SCM Metadata Store Metrics", context = OzoneConsts.OZONE)
 public final class SCMMetadataStoreMetrics implements MetricsSource {
 
+  private static final Logger LOG =
+      LoggerFactory.getLogger(SCMMetadataStoreMetrics.class);
+
   public static final String METRICS_SOURCE_NAME =
       SCMMetadataStoreMetrics.class.getSimpleName();
 
-  private static final MetricsInfo ESTIMATED_KEY_COUNT = Interns.info(
+  private static final MetricsInfo ESTIMATED_KEY_COUNT = info(
       "EstimatedKeyCount",
       "Tracked estimated key count of all column families");
 
-  private static final Map<String, MetricsInfo> ESTIMATED_KEY_COUNT_METRICS
```

Comment:
Now I understand - you would get a null pointer here, as this is a static definition. What about moving this to be an instance variable and build the `Interns.info` objects when this metrics object is created? That means we still have a cache of the names inside the object to avoid allocations each time the metrics are requested.

Reply:
Good idea. Updated the PR, please have a look.